test/perfdb.cpp

```diff
@@ -51,21 +51,25 @@ namespace tests {
 
 static boost::filesystem::path& exe_path()
 {
+    // NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
     static boost::filesystem::path exe_path;
     return exe_path;
 }
 
 static boost::optional<std::string>& thread_logs_root()
 {
+    // NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
     static std::mutex mutex;
     std::lock_guard<std::mutex> lock(mutex);
 
+    // NOLINTNEXTLINE (cppcoreguidelines-avoid-non-const-global-variables)
     static boost::optional<std::string> path(boost::none);
     return path;
 }
 
 static bool& full_set()
```

Comment:
Does anyone know if this feature is necessary? Or can it be removed?

Reply:
@shurale-nkn please ask @DrizztDoUrden

Reply:
@DrizztDoUrden ping

Reply:
I am not sure if anyone commonly uses this now, but in ancient times I have used it to debug PlainTextDb in a more stressful test. Is there any specific reason to remove it? It helps to triage db MP/MT failures, but one can just change values manually for the same result.

Reply:
@DrizztDoUrden @shurale-nkn It is used here, not Ok to remove:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/eae8583eb49db4dcc874d2eadee9051e70f9eae4/test/perfdb.cpp#L1290

Reply:
if true than true?)

Reply:
Read carefully: https://github.com/ROCmSoftwarePlatform/MIOpen/blob/eae8583eb49db4dcc874d2eadee9051e70f9eae4/test/perfdb.cpp#L1288-L1290

Reply:
I'm not sure is it a good decision. We are using static global tests::full_set() only in DbMultiProcessReadTest().Run(); and DbMultiProcessTest().Run(); which are called by the 6 lines below in same function, which means we can pass it as argument.

Reply:
The problem is that these ... ambiguous solutions, become the rule, not the exception, and they begin to be thoughtlessly copied.

Reply:
This decision is not good, but addressing it is out of the scope of this PR. /cc @DrizztDoUrden

Reply:
If you are talking about static variable specifically rather than --all mode for this test in general, I am 100% ok with changing it to a method parameter.